docs/source/CUSTOM-JOB.rst

```diff
@@ -0,0 +1,362 @@
+
+          nodeSelector:
+            consumer: "yes"
+
+          env:
+            DEBUG: "true"
+            INTERVAL: 1
+            QUEUE: "track"
+            CONSUMER_TYPE: "tracking"
+            EMPTY_QUEUE_TIMEOUT: 5
+            GRPC_TIMEOUT: 20
+            GRPC_BACKOFF: 3
+
+            REDIS_HOST: "redis"
+            REDIS_PORT: 26379
+            REDIS_TIMEOUT: 3
+
+            TF_HOST: "tf-serving"
+            TF_PORT: 8500
+            TF_TENSOR_NAME: "image"
+            TF_TENSOR_DTYPE: "DT_FLOAT"
+
+            AWS_REGION: '{{ env "AWS_REGION" | default "us-east-1" }}'
+            CLOUD_PROVIDER: '{{ env "CLOUD_PROVIDER" | default "aws" }}'
+            GKE_COMPUTE_ZONE: '{{ env "GKE_COMPUTE_ZONE" | default "us-west1-b" }}'
+
+            NUCLEAR_MODEL: "panoptic:3"
+            NUCLEAR_POSTPROCESS: "retinanet-semantic"
+
+            PHASE_MODEL: "resnet50_retinanet_20190813_all_phase_512:0"
+            PHASE_POSTPROCESS: "retinanet"
+
+            CYTOPLASM_MODEL:   "resnet50_retinanet_20190903_all_fluorescent_cyto_512:0"
+            CYTOPLASM_POSTPROCESS: "retinanet"
+
+            LABEL_DETECT_ENABLED: "true"
+            LABEL_DETECT_MODEL: "LabelDetection:0"
+            LABEL_RESHAPE_SIZE: 216
+            LABEL_DETECT_SAMPLE: 10
+
+            SCALE_DETECT_ENABLED: "true"
+            SCALE_DETECT_MODEL: "ScaleDetection:0"
+            SCALE_RESHAPE_SIZE: 216
+            SCALE_DETECT_SAMPLE: 10
+
+            DRIFT_CORRECT_ENABLED: "false"
+            NORMALIZE_TRACKING: "true"
+
+            TRACKING_MODEL: "tracking_model_benchmarking_757_step5_20epoch_80split_9tl:1"
+            TRACKING_SEGMENT_MODEL: "panoptic:3"
+            TRACKING_POSTPROCESS_FUNCTION: "retinanet"
+
+          secrets:
+            AWS_ACCESS_KEY_ID: '{{ env "AWS_ACCESS_KEY_ID" | default "NA" }}'
+            AWS_SECRET_ACCESS_KEY: '{{ env "AWS_SECRET_ACCESS_KEY" | default "NA" }}'
+            AWS_S3_BUCKET: '{{ env "AWS_S3_BUCKET" | default "NA" }}'
+            GKE_BUCKET: '{{ env "GKE_BUCKET" | default "NA" }}'
+
+4. Deploy your new helmfile to the cluster with:
+
+.. code-block:: bash
+
+    helmfile -l name=my-new-consumer sync
+
+.. |helm| raw:: html
+
+    <tt><a href="https://helm.sh/">helm</a></tt>
+
+.. |helmfile| raw:: html
+
+    <tt><a href="https://github.com/roboll/helmfile">helmfile</a></tt>
+
+Autoscaling custom consumers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To effectively scale your new consumer, some small edits will be needed in the following files:
+
+* |prometheus-redis-exporter.yaml|
+* |prometheus-operator.yaml|
+* |hpa.yaml|
+
+Generally, the consumer for each Redis queue is scaled relative to the amount of items in that queue. The work is tallied in the ``prometheus-redis-exporter``, the custom rule is defined in ``prometheus-operator``, and the Horizontal Pod Autoscaler is created and configured to use the new rule in the ``hpa.yaml`` file.
+
+1. |prometheus-redis-exporter.yaml|
```

Comment:
These are also in block quotes.

Reply:
See comment above about formatting issues on github.

Reply:
Fixed